dotCMS/src/main/java/com/dotcms/ai/api/OpenAITranslationService.java

```diff
@@ -0,0 +1,328 @@
+import com.dotmarketing.util.UtilMethods;
+import com.dotmarketing.util.VelocityUtil;
+import com.dotmarketing.util.json.JSONObject;
+import com.liferay.portal.model.User;
+import io.vavr.Lazy;
+import io.vavr.control.Try;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.velocity.context.Context;
+
+public class OpenAITranslationService extends AbstractTranslationService {
+
+    static final String AI_TRANSLATION_SYSTEM_PROMPT = "AI_TRANSLATION_SYSTEM_PROMPT";
+    static final String AI_TRANSLATION_USER_PROMPT = "AI_TRANSLATION_USER_PROMPT";
+    static final String AI_TRANSLATION_MODEL_KEY = "AI_TRANSLATION_MODEL";
+    static final String AI_TRANSLATIONS_MAX_TOKENS = "AI_TRANSLATIONS_MAX_TOKENS";
+    static final String AI_TRANSLATION_TEMPERATURE ="AI_TRANSLATION_TEMPERATURE";
+    static final String AI_TRANSLATION_RESPONSE_FORMAT ="AI_TRANSLATION_RESPONSE_FORMAT";
+
+    static int MAX_LANGUAGE_VARIABLE_CONTEXT = 1000;
+
+
+    public static final Lazy<TranslationService> INSTANCE = Lazy.of(()-> new OpenAITranslationService());
+
+
+
+
+    @Override
+    public String translateString(String toTranslate, Language from, Language to) throws TranslationException {
+        throw new DotRuntimeException("Not implemented");
+    }
+
+    @Override
+    public List<String> translateStrings(List<String> toTranslate, Language from, Language to)
+            throws TranslationException {
+        throw new DotRuntimeException("Not implemented");
+    }
+
+    @Override
+    public List<ServiceParameter> getServiceParameters() {
+        throw new DotRuntimeException("Not implemented");
+    }
+
+    @Override
+    public void setServiceParameters(List<ServiceParameter> params, String hostId) {
+        throw new DotRuntimeException("Not implemented");
+    }
+
+    @Override
+    public List<Contentlet> translateContent(Contentlet contentlet, List<Language> langs,
+            List<com.dotmarketing.portlets.structure.model.Field> oldFields,
+            User user) throws TranslationException {
+
+        return langs.stream().map(lang -> {
+                    try {
+                        return translateContent(contentlet, lang, oldFields, user);
+                    } catch (TranslationException e) {
+                        Logger.warnAndDebug(OpenAITranslationService.class, e.getMessage(), e);
+                        return null;
+                    }
+                })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+
+    }
+
+    @Override
+    public Contentlet translateContent(Contentlet contentlet, Language targetLanguage, List<com.dotmarketing.portlets.structure.model.Field> oldFields, User user)
+            throws TranslationException {
+        Language sourceLang = APILocator.getLanguageAPI().getLanguage(contentlet.getLanguageId());
```

Comment:
the method looks ok but it is a little bit longer, would be nice to make a refactor to split in into smaller methods